### DISCUSSION TITLE
How to create a custom toolbar button?

### DISCUSSION THREAD

**Discussion by jcao49** [jupyterlite/jupyterlite#589] [Q&A] [answered]
Hi, apologies if the answer is very obvious. I'm rather new to working with JupyterLab/JupyterLite and was wondering if there was a way to create a custom toolbar button that will execute some code when clicked similar to something like [this](https://github.com/jupyterlab/extension-examples/tree/master/toolbar-button) that is offered with JupyterLab? I have tried looking at the API documentation but am having some trouble understanding how to create such a feature. If someone could direct me in the right direction it would be very appreciated! Thanks :)

**Comment by bollwyvl** [comment, marked as answer]
Wherever possible, JupyterLite supports prebuilt labextensions directly.

So you can exactly follow that extension example (or the [cookiecutter](https://github.com/jupyterlab/extension-cookiecutter-ts)), and create a prebuilt extension. You _don't_ have to publish it on PyPI, though: [get it into the build](https://jupyterlite.readthedocs.io/en/latest/configuring.html#adding-extensions) by:
- installing the pip package in the environment where you do `jupyter lite build`
- point at its tarball
- point at its build directory

**Reply by bollwyvl** [reply]
Also, here's the canonical [extension developer guide](https://jupyterlab.readthedocs.io/en/stable/extension/extension_dev.html).

**Reply by jcao49** [reply]
Awesome, thank you for the clarification and links!!
